Mathlib/Algebra/Homology/Embedding/Restriction.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Algebra.Homology.Embedding.Basic
+import Mathlib.Algebra.Homology.Additive
+
+/-!
+# The restriction functor of an embedding of complex shapes
+
+Given `c` and `c'` complex shapes on two types, and `e : c.Embedding c'`,
+we define the restriction functor
+`e.restrictionFunctor C : HomologicalComplex C c' ⥤ HomologicalComplex C c`.
+
+-/
+
+open CategoryTheory Category Limits ZeroObject
+
+variable {ι ι' : Type*} {c : ComplexShape ι} {c' : ComplexShape ι'}
+
+namespace HomologicalComplex
+
+variable {C : Type*} [Category C] [HasZeroMorphisms C] [HasZeroObject C]
+  (K L M : HomologicalComplex C c') (φ : K ⟶ L) (φ' : L ⟶ M)
+  (e : c.Embedding c') [e.IsRelIff]
+
+/-- Given `K : HomologicalComplex C c'` and `e : c.Embedding c'` (satisfying `[e.IsRelIff]`),
+this is the homological complex in `HomologicalComplex C c` obtained by restriction. -/
+@[simps]
+def restriction : HomologicalComplex C c where
+  X i := K.X (e.f i)
+  d _ _ := K.d _ _
+  shape i j hij := K.shape _ _ (by simpa only [← e.rel_iff] using hij)
```

Comment:
```suggestion
  shape i j hij := K.shape _ _ ((e.rel_iff _ _).1.mt hij)
```

Reply:
Thanks for the other suggestions! I would rather keep using `simpa only` which uses the same number of lines (even though slightly more characters!), as it is slightly more readable IMHO.